Document/0x06g-Testing-Network-Communication.md

```diff
@@ -104,73 +89,76 @@ Starting from January 1 2017, Apple App Store review requires justification if o
+- Using low-level APIs (e.g. [`Network`](https://developer.apple.com/documentation/network)) or [`CFNetwork`](https://developer.apple.com/documentation/cfnetwork) to set up a custom HTTP connection.
+- Using a cross-platform framework (e.g. Flutter, Xamarin, ...), as these typically have their own implementations for HTTP libraries.
+
+All of the above cases must be carefully analyzed as a whole. For example, even if the app does not permit cleartext traffic in its Android Manifest or Network Security Configuration, it might actually still be sending HTTP traffic. That could be the case if it's using a low-level API (for which Network Security Configuration is ignored) or a badly configured cross-platform framework.
+
+For more information refer to the article "Preventing Insecure Network Connections" in the [Apple Developer Documentation](https://developer.apple.com/documentation/security/preventing_insecure_network_connections).
+
+### Static Analysis
 
-================================================================================
+#### Testing Network API Usage
 
-Allowing Arbitrary Loads
+Start by identifying the network APIs used by the app and see if it mainly uses high-level networking APIs.
 
----
-Allow All Loads
-Result : PASS
----
+> **Apple Recommendation: Prefer High-Level Frameworks in Your App**: ATS doesn’t apply to calls your app makes to lower-level networking interfaces like the Network framework or CFNetwork. In these cases, you take responsibility for ensuring the security of the connection. You can construct a secure connection this way, but mistakes are both easy to make and costly. It’s typically safest to rely on the URL Loading System instead.
 
-================================================================================
+If it uses low-level APIs such as [`Network`](https://developer.apple.com/documentation/network)) or [`CFNetwork`](https://developer.apple.com/documentation/cfnetwork) you should further investigate if they are being used securely. For apps using cross-platform frameworks (e.g. Flutter, Xamarin, ...) you should analyze if they're being configured and used securely according to the framework's best practices.
 
-Configuring TLS exceptions for www.example.com
+#### Testing Network API Usage
 
----
-TLSv1.3
-2019-01-15 09:39:27.892 nscurl[11459:5126999] NSURLSession/NSURLConnection HTTP load failed (kCFStreamErrorDomainSSL, -9800)
-Result : FAIL
----
+You should verify if the app allows cleartext traffic. If the source code is available, open then `Info.plist` file in the application bundle directory and look for any exceptions within the [`NSAppTransportSecurity`](https://developer.apple.com/documentation/bundleresources/information_property_list/nsapptransportsecurity) optional key.
```

Comment:
```suggestion
You should verify if the app allows cleartext traffic. Open the `Info.plist` file in the application bundle directory and look for any exceptions within the [`NSAppTransportSecurity`](https://developer.apple.com/documentation/bundleresources/information_property_list/nsapptransportsecurity) optional key.
```
No source code is required. You can extract it from the ipa or the installed application on a jailbroken iphone.

Reply:
I see there's a section below that has this. Shouldn't they just be merged? Seems very unnecessary to split them up.